### ISSUE TITLE
Consider a opt-in consent model

### ISSUE THREAD

**Issue by ocdtrekkie** [ericdfields/threadstadon#7]
There's a certain crowd of fediverse admins explicitly opposed to any tools that collect data from even the public Mastodon API. Respecting #nobot #noindex #noarchive is *good*, but still lives in the land of opt-out behavior instead of opt-in behavior (i.e. consent).

Would you consider requiring an unrollable thread to either contain something like "#unroll" either in a toot from the original author on the thread for just that thread, or on their profile to enable support for all of their posts?

**Comment by ericdfields** [comment]
It's a change I'd rather see on the protocol level. 

I literally just learned about adding #no* tags in bios today. This seems to be an implied community standard that I was unaware of this whole time. The same would be true of #unroll opt-in to the rest of the fediverse. 

I'd rather just shut this down in favor of a more readable standalone status view, as proposed here: https://github.com/mastodon/mastodon/issues/20865

Is this actually a terrible project and the people that find is useful don't know enough??

**Comment by ocdtrekkie** [comment]
I think the concept of integrating it directly into Mastodon would be far preferable to a lot of people.

I don't think it's a terrible project, but there is a subset of instance admins that are extremely hostile to tools which "scrape" or "index" the fediverse in any way that is not explicitly opt-in. Client-side front-ends seems to largely be accepted, so another potential option would be to have the unroller generate the thread at load time, and not cache/store it at all. But I suspect that will be more painful especially when the servers are under load.

**Comment by ericdfields** [comment]
Yeah the day I finished it last week, I stumbled upon some good client side examples of the same thing. “Why’d I even database??” 

But the core functionality is reformatting toots once for reading/sharing. Doing it every time on client will one day drag down a server. But maybe not? I make one API call for status (parent) and one for context (descendants). I *could* provide a short lived cache that expires in 8 hours until it is requested to be rebuilt by a new client?

**Comment by ocdtrekkie** [comment]
I mean, conceivably that model shouldn't ever be more server-intensive than "you linked people to the Mastodon thread directly", no? If it was a temporary cache, it would likely count more as a client app than a scraper (to people who need the distinction) and would effectively solve #8 for free.

**Comment by ericdfields** [comment]
Yeah it would.

Thanks for taking the time here @ocdtrekkie, but I wasn't ready for a side project to be the bane of a server admin's existence. I'm just gonna pull the plug.

**Comment by ocdtrekkie** [comment]
That is pretty much the goal of folks who tend to immediately encourage blocking these. :/

Not sure if contributing to Mastodon proper is in your wheelhouse or not, but it would probably be welcomed.

**Comment by mediaformat** [comment]
I think this could work, as opt-in if the thread creator somehow agreed to it. Alternately the app could work privately as a thread reader app that users could login to.
